src/entities/Map/utils.ts

```diff
@@ -0,0 +1,49 @@
+import { SceneStats, SceneStatsMap } from "../../api/DataTeam"
+import { HotScene } from "../Place/types"
+import { AggregateCoordinatePlaceAttributes } from "./types"
+
+export function placesWithCoordinatesAggregates(
+  places: AggregateCoordinatePlaceAttributes[],
+  hotScenes: HotScene[],
+  sceneStats: SceneStatsMap,
+  options?: {
+    withRealmsDetail: boolean
+  }
+) {
+  const placesWithAggregates: Record<
+    string,
+    AggregateCoordinatePlaceAttributes
+  > = {}
+
+  for (const place of places) {
+    let stats: SceneStats | undefined = sceneStats[place.base_position]
+
+    if (!stats) {
+      const statsPosition = (place.positions || []).find(
+        (position) => sceneStats[position]
+      )
+      if (statsPosition) {
+        stats = sceneStats[statsPosition]
+      }
+    }
```

Comment:
Would you mind avoiding [nest](https://www.youtube.com/watch?v=CFRhGnuXG-4) code? 
Maybe something like
```suggestion
    let statsPosition
    if (!stats) {
      statsPosition = (place.positions || []).find(
        (position) => sceneStats[position]
      )
    }
    if (!stats && statsPosition) {
        stats = sceneStats[statsPosition]
    }
```